enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -4476,6 +4476,103 @@ class AdjointGenerator
         return;
       }
 
+      // Functions that only modify pointers and don't allocate memory,
+      // needs to be run on shadow in primal
+      if (funcName ==
+          "_ZSt29_Rb_tree_insert_and_rebalancebPSt18_Rb_tree_node_baseS0_RS_") {
```

Comment:
This seems rather unfortunate (and orthogonal to this PR)

Could we solve this by using a bit code version of libstdc++?